### PR TITLE
Js memoize more

### PR DESCRIPTION
**Pull request by jsingle**
@thieman memoizing a couple more functions to cut down args we pass around

### CODE REVIEW

File: dusty/systems/docker/testing_image.py [hunk 1]
```diff
@@ -58,12 +61,14 @@ def _ensure_image_exists(docker_client, image_name):
         repo, tag = split[0], 'latest' if len(split) == 1 else split[1]
         docker_client.pull(repo, tag, insecure_registry=True)
 
-def _make_installed_requirements_image(docker_client, base_image_tag, command, image_name, volumes):
+def _make_installed_requirements_image(base_image_tag, command, image_name, volumes):
+    docker_client = get_docker_client()
+    print docker_client.wait()
```
Comment (thieman) [inline dusty/systems/docker/testing_image.py]: :dolphin:

Comment (jsingle) [inline dusty/systems/docker/testing_image.py]: haha test struggles

**Comment by thieman** [comment]
yay cleaner code

**Comment by jsingle** [comment]
...and failing tests that passed locally

**Pull request closed without merging**